mne/preprocessing/faster.py

```diff
@@ -0,0 +1,368 @@
+
+    # Limit power spectrum to upper frequencies
+    ps = ps[:, np.searchsorted(fs, 25):np.searchsorted(fs, 45)]
+
+    # Compute mean gradients
+    return np.mean(np.diff(ps), axis=1)
+
+
+def _deviation(data):
+    """Computes the deviation from mean for each channel in a set of epochs.
+
+    This is not implemented as a lambda function, because the channel means
+    should be cached during the computation.
+    
+    Parameters
+    ----------
+    data : 3D numpy array
+        The epochs (#epochs x #channels x #samples).
+
+    Returns
+    -------
+    dev : 1D numpy array
+        For each epoch, the mean deviation of the channels.
+    """
+    ch_mean = np.mean(data, axis=2)
+    return ch_mean - np.mean(ch_mean, axis=0)
+
+
+def faster_bad_channels(epochs, picks=None, thres=3, use_metrics=None):
+    """Implements the first step of the FASTER algorithm.
+    
+    This function attempts to automatically mark bad EEG channels by performing
+    outlier detection. It operated on epoched data, to make sure only relevant
+    data is analyzed.
+
+    Parameters
+    ----------
+    epochs : Instance of Epochs
+        The epochs for which bad channels need to be marked
+    picks : list of int | None
+        Channels to operate on. Defaults to EEG channels.
+    thres : float
+        The threshold value, in standard deviations, to apply. A channel
+        crossing this threshold value is marked as bad. Defaults to 3.
+    use_metrics : list of str
+        List of metrics to use. Can be any combination of:
+            'variance', 'correlation', 'hurst', 'kurtosis', 'line_noise'
+        Defaults to all of them.
+
+    Returns
+    -------
+    bads : list of str
+        The names of the bad EEG channels.
+    """
+    metrics = {
+        'variance':    lambda x: np.var(x, axis=1),
+        'correlation': lambda x: nanmean(
+                           np.ma.masked_array(
+                               np.corrcoef(x),
+                               np.identity(len(x), dtype=bool)
+                           ),
+                           axis=0),
+        'hurst':       lambda x: _hurst(x),
+        'kurtosis':    lambda x: kurtosis(x, axis=1),
+        'line_noise':  lambda x: _freqs_power(x, epochs.info['sfreq'],
+                                              [50, 60]),
+    }
+
+    if picks is None:
+        picks = pick_types(epochs.info, meg=False, eeg=True, exclude=[])
+    if use_metrics is None:
+        use_metrics = metrics.keys()
+
+    # Concatenate epochs in time
+    data = epochs.get_data()
+    data = data.transpose(1, 0, 2).reshape(data.shape[1], -1)
+    data = data[picks]
+
+    # Find bad channels
+    bads = []
+    for m in use_metrics:
+        s = metrics[m](data)
+        b = [epochs.ch_names[picks[i]] for i in find_outliers(s, thres)]
+        logger.info('Bad by %s:\n\t%s' % (m, b))
+        bads.append(b)
+
+    return np.unique(np.concatenate(bads)).tolist()
```

Comment:
but for regular use, you want a plain list. The bads detected by each method is already listed in the log... would plotting the bads detected by each method be a frequent enough use case to warrant an extra parameter?